assets/users/reducers.js

```diff
@@ -102,6 +102,8 @@ export default function userReducer(state = initialState, action) {
             } else {
                 user.products = (user.products || []).filter((product) => product._id !== productId);
             }
+        } else if (field.includes('selectAllBtn')) {
+            user.products = state.products.filter((product)=> user.sections[product.product_type]);
```

Comment:
not sure about the values you're assigning here, on user we only store `_id` and `section`, don't we?

Reply:
We preserved `product._id` in `user.products` and used logic if user section is true _(eg:- agenda)_ so that in this case all agenda **products** are selected from the state and then assigned to 'user.products'.

Reply:
yep sure I'm rather wondering if there should be as well:
```ts
.map((product) => {_id: product._id, section: product.product_type})
```
because we don't store the whole product objects in `user.products`